include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -152,6 +152,84 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<Swizzle,Dither,Nor
+
+					const core::vectorSIMDu32 phaseCount = getPhaseCount(state->inExtentLayerCount, state->outExtentLayerCount, state->inImage->getCreationParameters().type);
+
+					const core::vectorSIMDu32 axisOffsets = getPhaseSupportLUTAxisOffsets(phaseCount, scaledKernelX, scaledKernelY, scaledKernelZ);
+
+					const core::vectorSIMDf fInExtent(state->inExtentLayerCount);
+					const core::vectorSIMDf fOutExtent(state->outExtentLayerCount);
+					const auto fScale = fInExtent.preciseDivision(fOutExtent);
+
+					// a dummy load functor
+					// does nothing but fills up the `windowSample` with 1s (identity) so we can preserve the value of kernel
+					// weights when eventually `windowSample` gets multiplied by them later in
+					// `CFloatingPointSeparableImageFilterKernelBase<CRTP>::sample_functor_t<PreFilter,PostFilter>::operator()`
+					// this exists only because `evaluateImpl` expects a pre filtering step.
+					auto dummyLoad = [](value_type* windowSample, const core::vectorSIMDf& unused0, const core::vectorSIMDi32& unused1, const IImageFilterKernel::UserData* unused2) -> void
+					{
+						for (auto h = 0; h < MaxChannels; h++)
+							windowSample[h] = value_type(1);
+					};
+
+					// There's no real reason that this should have `MaxChannels` number of elements except for the fact that
+					// `CFloatingPointSeparableImageFilterKernelBase<CRTP>::sample_functor_t<PreFilter,PostFilter>::operator()` expects that.
+					// So this, for now, just replicates the same value in all 4 channels.
+					value_type kernelWeight[MaxChannels];
+					// actually used to put values in the LUT
+					auto dummyEvaluate = [&kernelWeight](const value_type* windowSample, const core::vectorSIMDf& unused0, const core::vectorSIMDi32& unused1, const IImageFilterKernel::UserData* unused2) -> void
+					{
+						for (auto h = 0; h < MaxChannels; h++)
+							kernelWeight[h] = windowSample[h];
+					};
+
+					auto computeForAxis = [state, &fScale, &phaseCount, &dummyLoad, &dummyEvaluate, &kernelWeight, &axisOffsets](const asset::IImage::E_TYPE axis, const auto& kernel)
+					{
+						if (axis > state->inImage->getCreationParameters().type)
+							return;
+
+						const auto windowSize = kernel.getWindowSize()[axis];
+
+						IImageFilterKernel::ScaleFactorUserData scale(1.f / fScale[axis]);
+						const IImageFilterKernel::ScaleFactorUserData* otherScale = IImageFilterKernel::ScaleFactorUserData::cast(state->kernelX.getUserData());
+
+						if (otherScale)
+						{
+							for (auto k = 0; k < MaxChannels; k++)
+								scale.factor[k] *= otherScale->factor[k];
+						}
+
+						value_type* phaseSupportLUTPixel = reinterpret_cast<value_type*>(state->scratchMemory + getPhaseSupportLUTByteOffset(state) + axisOffsets[axis]);
+						for (uint32_t i = 0u; i < phaseCount[axis]; ++i)
+						{
+							core::vectorSIMDf tmp;
+							tmp[axis] = float(i) + 0.5f;
+
+							core::vectorSIMDi32 windowCoord;
```

Comment:
just use an `int32_t`

Reply:
I would still need to pass a `core::vectorSIMDi32` in the loop below which the LOC:
```
kernel.evaluateImpl(dummyLoad, dummyEvaluate, kernelWeight, tmp, windowCoord, &scale);
```

Reply:
yeah, you should create the `core::vectorSIMDi32` on demand there, within the loop, compiler might optimize a few registers away.

Reply:
Resolved in #343.